x-pack/plugin/blob-cache/src/test/java/org/elasticsearch/blobcache/common/ProgressListenableActionFutureTests.java

```diff
@@ -233,9 +233,46 @@ public void testListenerCalledImmediatelyWhenProgressReached() {
         assertThat(future.isDone(), is(true));
     }
 
+    public void testLongConsumerCalledOnProgressUpdate() {
+        // min length of 2 to have at least one progress update before reaching the end
+        long length = randomLongBetween(2L, ByteSizeUnit.TB.toBytes(1L));
+        long start = randomLongBetween(Long.MIN_VALUE, Long.MAX_VALUE - length);
```

Comment:
The code does not prevent negative `start` values. But it seems rather odd to allow it. I wonder whether we should tighten it up in a separate PR to assert `start >= 0`?